src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -286,48 +413,50 @@ void DecisionTree<FitnessFunction,
       numChildren = NumericSplit::NumChildren(classProbabilities, *this);
 
     // Calculate all child assignments.
-    arma::Col<size_t> childAssignments(data.n_cols);
+    arma::Row<size_t> childAssignments(data.n_cols);
```

Comment:
We are only using elements `begin` to `begin + count` here, I think we could save memory by reducing this to have size `count` instead of `data.n_cols`.